File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.java
class Solution {
    public int longestSubsequence(int[] nums) {
        int n=nums.length;
        boolean zero=true;
        for(int num:nums){
            if(num!=0){
                zero=false;
                break;
            }
        }
        if(zero){
            return 0;
        }
        int x=0;
        for(int num:nums){
            x^=num;
        }
        return x!=0 ? n:n-1;
    }
}

File: 4033-longest-subsequence-with-non-zero-bitwise-xor/longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        ele=0;xor=True
        for num in nums:
            ele=(ele^num)
            if num!=0:
                xor=False
        if xor:
            return 0
        return len(nums)-1 if ele==0 else len(nums)
        